eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,173 @@
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+/**
+ * Check Style Rule: Good Logging Practice
+ *  <ol>
+ *      <li>Non-static Logger instance.</li>
+ *      <li>Logger variable name should be 'logger' in public class and not exist in an implementation package.</li>
+ *      <li>Should not use any external logger class, only use ClientLogger. No slf4j, log4j, or other logging imports are allowed.</li>
+ *      <li>Should refer to the containing class in its constructor:
+ *      public class A {
+ *          private ClientLogger logger = new ClientLogger(A.class);
+ *      }
+ *      </li>
+ *  </ol>
+ */
+public class GoodLoggingCheck extends AbstractCheck {
+    private static final String CLIENT_LOOGER_PATH = "com.azure.core.util.logging.ClientLogger";
+    private static final String CLIENT_LOGGER = "ClientLogger";
+    private static final String SLF4J = "org.slf4j";
+    private static final String LOG4J = "org.apache.logging.log4j";
+
+    private static final String LOGGER_NAME_ERROR = "ClientLogger instance naming: use ''%s'' instead of ''%s'' for consistency.";
+    private static final String STATIC_LOGGER_ERROR = "Reference to ClientLogger should not be static: remove static modifier.";
+    private static final String NOT_CLIENT_LOGGER_ERROR = "Do not use external logger class. Use ''%s'' as a logging mechanism instead of ''%s''.";
+
+    private boolean hasSlf4jImported;
+    private boolean hasLog4jimported;
+    private boolean hasClientLoggerImported;
+    private String loggerRequiredName;
+    private String className;
+
+    /**
+     * Setter to specifies valid identifiers
+     * @param loggerRequiredName the variable name of logger used
+     */
+    public void setLoggerName(String loggerRequiredName) {
+        this.loggerRequiredName = loggerRequiredName;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.IMPORT,
+            TokenTypes.LITERAL_CLASS,
+            TokenTypes.LITERAL_NEW,
+            TokenTypes.VARIABLE_DEF
+        };
+    }
+
+    @Override
+    public void finishTree(DetailAST ast) {
+        hasSlf4jImported = false;
+        hasLog4jimported = false;
+        hasClientLoggerImported = false;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.IMPORT:
+                String importClassPath = FullIdent.createFullIdentBelow(ast).getText();
+                hasClientLoggerImported |= importClassPath.equals(CLIENT_LOOGER_PATH);
+                hasSlf4jImported |= importClassPath.startsWith(SLF4J);
+                hasLog4jimported |= importClassPath.startsWith(LOG4J);
```

Comment:
You shouldn't use the pipe equal operator here. Just do standard assignment operation.